scilpy/io/utils.py

```diff
@@ -589,12 +591,23 @@ def is_header_compatible_multiple_files(parser, list_files,
 
     for filepath in list_files:
         _, in_extension = split_name_with_nii(filepath)
-        if in_extension not in ['.trk', '.nii', '.nii.gz']:
+        if in_extension not in ['.tck', '.trk', '.nii', '.nii.gz']:
             parser.error('{} does not have a supported extension'.format(
                 filepath))
 
     for curr in list_files[1:]:
-        if not is_header_compatible(list_files[0], curr):
+        compatible = True
+        if in_extension == '.tck':
+            # TCKs do not have a header, so compare with the reference's
+            # header.
+            _, curr_extension = split_name_with_nii(curr)
+            # If comparing two TCKs, the same reference will be used by
+            # both so they will be compatible.
+            if curr_extension != '.tck':
+                compatible = is_header_compatible(reference, curr)
+        else:
+            compatible = is_header_compatible(list_files[0], curr)
```

Comment:
So here it assumes list_files[0] is a file that has an header to check against.

I think a better way of doing it would be that in the first for loop (`for filepath in list_files:`) you would have to find which of the input has a header (only `['.trk', '.nii', '.nii.gz'])` and remember which position it was and use that position instead of 0 in `list_files[0]`

The idea is that if you have:
- Only TCK: Do not check header since they dont have any
- A mix of TCK and TRK: Check the TRKs and the --reference so they match
- Only TRK: Check the TRKs so they match

So the thing missing in this part is that in_extension should now be used to detect if you have a least one TRK, and if you do have one its position should be remember for the check just after.

Reply:
Sure !

In the first loop, I'll actually gather "headers" (be it nifti/trk headers or a tck's reference's header) and then I'll compare them as usual. Should be clean enough.